twilight-gateway/src/shard.rs

```diff
@@ -1231,19 +1232,7 @@ impl Shard {
                     source
                 })?,
         );
-
-        if self.session().is_some() {
-            // Defer sending a Resume event until Hello has been received to
-            // guard against the first message being a websocket close message
-            // (causing us to miss replayed dispatch events).
-            // We also set/reset the ratelimiter upon receiving Hello, which
-            // means sending anything before then will not be recorded by the
-            // ratelimiter.
-            self.status = ConnectionStatus::Resuming;
-        } else {
-            self.status = ConnectionStatus::Identifying;
-        }
```

Comment:
When removing this `ConnectionStatus::Resuming` will essentially never be used therefore making `is_resuming` also unnecessary 🤔 
Maybe instead of doing it like this we should change how `is_identified` or how sending messages work.

Reply:
`Resuming` is set until `Resumed` is received.

The Discord thread discusses some alternative solution but ultimately this was the _least_ breaking change I could come up with. I'll gladly switch if you have an alternative patch that you feel is better.